From: Miaohe Lin <linmiaohe@huawei.com>
Subject: Revert "hugetlb: make hugetlb depends on SYSFS or SYSCTL"
Date: Tue, 26 Aug 2025 11:09:55 +0800

Commit f8142cf94d47 ("hugetlb: make hugetlb depends on SYSFS or SYSCTL")
added dependency on SYSFS or SYSCTL but hugetlb can be used without SYSFS
or SYSCTL.  So this dependency is wrong and should be removed.

For users with CONFIG_SYSFS or CONFIG_SYSCTL on, there should be no
difference.  For users have CONFIG_SYSFS and CONFIG_SYSCTL both
undefined, hugetlbfs can still works perfectly well through cmdline
except a possible kismet warning[1] when select CONFIG_HUGETLBFS. 
IMHO, it might not worth a backport.

This reverts commit f8142cf94d4737ea0c3baffb3b9bad8addcb9b6b.  It
overlooked the scenario of using hugetlb through boot parameters when
it was submitted.

Link: https://lkml.kernel.org/r/20250826030955.2898709-1-linmiaohe@huawei.com
Link: https://lore.kernel.org/all/5c99458f-4a91-485f-8a35-3618a992e2e4@csgroup.eu/ [1]
Reported-by: kernel test robot <lkp@intel.com>
Closes: https://lore.kernel.org/oe-kbuild-all/202508222032.bwJsQPZ1-lkp@intel.com/
Signed-off-by: Miaohe Lin <linmiaohe@huawei.com>
Cc: David Hildenbrand <david@redhat.com>
Cc: Muchun Song <muchun.song@linux.dev>
Cc: Oscar Salvador <osalvador@suse.de>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 fs/Kconfig |    1 -
 1 file changed, 1 deletion(-)

--- a/fs/Kconfig~revert-hugetlb-make-hugetlb-depends-on-sysfs-or-sysctl
+++ a/fs/Kconfig
@@ -250,7 +250,6 @@ config ARCH_SUPPORTS_HUGETLBFS
 menuconfig HUGETLBFS
 	bool "HugeTLB file system support"
 	depends on ARCH_SUPPORTS_HUGETLBFS
-	depends on (SYSFS || SYSCTL)
 	select MEMFD_CREATE
 	select PADATA if SMP
 	help
_
